tests/json_test.cpp

```diff
@@ -170,5 +170,16 @@ void JsonUnsortedArrayTest() {
   TEST_NOTNULL(monster->testarrayoftables()->LookupByKey("ccc"));
 }
 
+void JsonUnionStructTest(const std::string &tests_data_path) {
+  std::string schemafile;
+  TEST_EQ(flatbuffers::LoadFile(
+              (tests_data_path + "json_union_struct_test.fbs").c_str(), false,
+              &schemafile),
+          true);
+  flatbuffers::Parser parser;
+  parser.opts.lang_to_generate = IDLOptions::kJson;
+  TEST_EQ(parser.Parse(schemafile.c_str()), true);
+}
```

Comment:
The above is fine for the parsing, but don't you want to generate the JSON too and compare that it generates valid JSON?

Reply:
Hi @dbaileychess,
I agree, the generating text back should be added.
Is it what you expect?

Reply:
@dbaileychess I added test as you wished. Could you check if it’s what you want?